nachos/src/main/java/com/hootsuite/nachos/chip/UpdateableChipSpan.java

```diff
@@ -0,0 +1,552 @@
+ * through the various setter methods available in this class.
+ *     The basic structure of the chip is the following:
+ * For chips with the icon on right:
+ * <pre>
+ *
+ *                                  (chip vertical spacing / 2)
+ *                  ----------------------------------------------------------
+ *                |                                                            |
+ * (left margin)  |  (padding edge)   text   (padding between image)   icon    |   (right margin)
+ *                |                                                            |
+ *                  ----------------------------------------------------------
+ *                                  (chip vertical spacing / 2)
+ *
+ *      </pre>
+ * For chips with the icon on the left (see {@link #setShowIconOnLeft(boolean)}):
+ * <pre>
+ *
+ *                                  (chip vertical spacing / 2)
+ *                  ----------------------------------------------------------
+ *                |                                                            |
+ * (left margin)  |   icon  (padding between image)   text   (padding edge)    |   (right margin)
+ *                |                                                            |
+ *                  ----------------------------------------------------------
+ *                                  (chip vertical spacing / 2)
+ *     </pre>
+ */
+public class UpdateableChipSpan extends ImageSpan implements Chip {
+
+    private static final float SCALE_PERCENT_OF_CHIP_HEIGHT = 0.70f;
+    private static final boolean ICON_ON_LEFT_DEFAULT = true;
+
+    private int[] mStateSet = new int[]{};
+
+    private String mEllipsis;
+
+    private ColorStateList mDefaultBackgroundColor;
+    private ColorStateList mBackgroundColor;
+    private int mTextColor;
+    private int mIconBackgroundColor;
+
+    private int mTextSize = -1;
+    private int mPaddingEdgePx;
+    private int mPaddingBetweenImagePx;
+    private int mLeftMarginPx;
+    private int mRightMarginPx;
+    private int mMaxAvailableWidth = -1;
+
+    private CharSequence mText;
+    private String mTextToDraw;
+
+    private StateListDrawable mIcon;
+    private boolean mShowIconOnLeft = ICON_ON_LEFT_DEFAULT;
+
+    private int mChipVerticalSpacing = 0;
+    private int mChipHeight = -1;
+    private int mChipWidth = -1;
+    private int mIconWidth;
+
+    private int mCachedSize = -1;
+
+    private Object mData;
+
+    /**
+     * Constructs a new ChipSpan.
+     *
+     * @param context a {@link Context} that will be used to retrieve default configurations from resource files
+     * @param text    the text for the ChipSpan to display
+     * @param icon    an optional icon (can be {@code null}) for the ChipSpan to display
+     */
+    public UpdateableChipSpan(@NonNull Context context, @NonNull CharSequence text, @Nullable StateListDrawable icon, Object data) {
```

Comment:
❓  This constructor is the essentially the same as the original one (just below this constructor). What are benefits are you getting by creating this new constructor? Can they be merged?